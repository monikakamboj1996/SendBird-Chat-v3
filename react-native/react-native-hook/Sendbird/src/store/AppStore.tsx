import AsyncStorage from '@react-native-async-storage/async-storage';
import { createStore, applyMiddleware } from 'redux';

import { persistStore, persistReducer } from 'redux-persist';
import thunk from 'redux-thunk';
import RootReducer from '../reducer/RootReducer';

// Middleware: Redux Persist Config
const persistConfig = {
  // Root
  key: 'root',
  // Storage Method (React Native)
  storage: AsyncStorage,
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, RootReducer);

// Redux: Store
const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);
// Exports
export { store, persistor };
