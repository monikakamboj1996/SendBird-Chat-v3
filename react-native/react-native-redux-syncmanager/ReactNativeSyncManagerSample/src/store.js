import {createStore, compose, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducer';

export const store = createStore(reducers, {}, compose(applyMiddleware(thunk)));
