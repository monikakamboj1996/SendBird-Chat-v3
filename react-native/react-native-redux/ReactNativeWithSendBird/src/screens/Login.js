import React, { Component } from 'react';
import { View, Text, TextInput, Image } from 'react-native';
import { connect } from 'react-redux';
import { initLogin, sendbirdLogin } from '../actions';
import { sbRegisterPushToken } from '../sendbirdActions';
import { NavigationActions, StackActions } from 'react-navigation';
import { Button, Spinner } from '../components';
import firebase from 'react-native-firebase';

class Login extends Component {
  static navigationOptions = {
    header: null
  };

  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      userId: '',
      nickname: ''
    };
  }

  componentDidMount() {
    this.props.initLogin();
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    let { user, error } = this.props;
    if (user && user !== prevProps.user) {
      firebase
        .messaging()
        .getToken()
        .then(pushToken => {
          if (pushToken) {
            sbRegisterPushToken(pushToken)
              .then(res => {})
              .catch(err => {});
          }
        });
      const resetAction = StackActions.reset({
        index: 0,
        actions: [NavigationActions.navigate({ routeName: 'Menu' })]
      });
      this.setState({ userId: '', nickname: '', isLoading: false }, () => {
        this.props.navigation.dispatch(resetAction);
      });
    }

    if (error && error !== prevProps.error) {
      this.setState({ isLoading: false });
    }
  }

  _onUserIdChanged = userId => {
    this.setState({ userId });
  };

  _onNicknameChanged = nickname => {
    this.setState({ nickname });
  };

  _onButtonPress = () => {
    const { userId, nickname } = this.state;
    this.setState({ isLoading: true }, () => {
      this.props.sendbirdLogin({ userId, nickname });
    });
  };

  render() {
    return (
      <View style={styles.containerStyle}>
        <Spinner visible={this.state.isLoading} />
        <View style={styles.logoViewStyle}>
          <Image style={{ width: 150, height: 150 }} source={require('../img/icon_sb_512.png')} />
          <Text style={styles.logoTextTitle}>SendBird</Text>
          <Text style={styles.logoTextSubTitle}>React Native</Text>
        </View>

        <View style={styles.inputViewStyle}>
          <TextInput
            label="User ID"
            placeholder="User ID"
            style={styles.inputStyle}
            value={this.state.userId}
            duration={100}
            autoCorrect={false}
            maxLength={16}
            underlineColorAndroid="transparent"
            onChangeText={this._onUserIdChanged}
          />
        </View>

        <View style={styles.inputViewStyle}>
          <TextInput
            label="Nickname"
            placeholder="Nickname"
            style={styles.inputStyle}
            value={this.state.nickname}
            duration={100}
            autoCorrect={false}
            maxLength={16}
            underlineColorAndroid="transparent"
            onChangeText={this._onNicknameChanged}
          />
        </View>

        <View style={styles.buttonStyle}>
          <Button
            title="CONNECT"
            buttonStyle={{ backgroundColor: '#6e5baa' }}
            onPress={this._onButtonPress}
            disabled={this.state.isLoading}
          />
        </View>

        <Text style={styles.errorTextStyle}>{this.props.error}</Text>

        <View style={[styles.footerViewStyle]}>
          <Text style={styles.footerTextStyle}>Sample UI v3.0.0 / SDK v.3.0.99</Text>
        </View>
      </View>
    );
  }
}

function mapStateToProps({ login }) {
  const { error, user } = login;
  return { error, user };
}

export default connect(
  mapStateToProps,
  { initLogin, sendbirdLogin }
)(Login);

const styles = {
  containerStyle: {
    backgroundColor: '#fff',
    flex: 1
  },
  logoViewStyle: {
    marginTop: 35,
    marginBottom: 5,
    alignItems: 'center'
  },
  logoTextTitle: {
    color: '#7d62d9',
    fontSize: 30,
    fontWeight: '600'
  },
  logoTextSubTitle: {
    color: '#7d62d9',
    fontSize: 13,
    fontWeight: '500'
  },
  inputViewStyle: {
    borderWidth: 1,
    borderColor: '#ddd',
    borderRadius: 5,
    paddingLeft: 8,
    paddingRight: 8,
    marginLeft: 28,
    marginRight: 28,
    marginTop: 8
  },
  inputStyle: {
    fontSize: 13,
    backgroundColor: '#fff'
  },
  buttonStyle: {
    paddingLeft: 12,
    paddingRight: 12,
    marginTop: 50
  },
  errorTextStyle: {
    alignSelf: 'center',
    fontSize: 12,
    color: '#e03131'
  },
  footerViewStyle: {
    paddingLeft: 28,
    paddingRight: 28,
    marginTop: 15,
    flexDirection: 'column'
  },
  footerTextStyle: {
    alignSelf: 'center',
    fontSize: 12,
    color: '#8e8e8e'
  }
};
