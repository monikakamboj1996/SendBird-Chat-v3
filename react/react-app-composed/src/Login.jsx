import React from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Avatar from '@material-ui/core/Avatar';

import {
  useHistory,
} from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  main: {
    backgroundColor: 'white',
    textAlign: 'center',
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
    backgroundColor: '#6e5baa',
    '&:hover': {
      backgroundColor: '#825deb',
    },
  },
  checkbox: {
    '&.Mui-checked': {
      color: '#825deb',
    },
  },
  input: {
    '& .Mui-focused': {
      color: '#825deb',
    },
    '& .MuiOutlinedInput-root': {
      '&.Mui-focused fieldset': {
        borderColor: '#825deb',
      },
    }
  },
  avatar: {
    marginTop: 20,
    width: 127 / 2,
    height: 150 / 2,
    borderRadius: 0,
  },
}));

export default function SignIn({ onSubmit }) {
  const classes = useStyles();
  const history = useHistory();
  return (
    <Container component="main" maxWidth="xs" className={classes.main}>
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar
          alt="Sendbird Logo"
          src="https://dxstmhyqfqr1o.cloudfront.net/symbol/Sendbird_Symbol_PNG/Sendbird_Symbol_RGB.png"
          className={classes.avatar}
        />
        <Typography component="h1" variant="h5">
          Sendbird | UIKit Sample
        </Typography>
        <form
          className={classes.form}
          onSubmit={e => {
            e.preventDefault();
            onSubmit({
              userId: e.target.userId.value,
              nickname: e.target.nickname.value,
              theme: e.target.theme.checked ? 'dark' : 'light',
            });
            history.push('/chat');;
          }}
        >
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="userId"
            label="User Id"
            name="userId"
            className={classes.input}
            autoFocus
          />
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="nickname"
            label="Nick Name"
            id="nickname"
            className={classes.input}
          />
          <FormControlLabel
            control={<Checkbox value="dark" color="primary" name="theme" className={classes.checkbox} />}
            label="Apply dark theme"
          />
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            Login
          </Button>
        </form>
      </div>
      <Typography variant="body1">
        Start chatting on Sendbird by choosing your username and nickname.
      </Typography>
    </Container>
  );
}
