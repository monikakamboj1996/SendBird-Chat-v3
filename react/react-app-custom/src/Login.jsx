import React from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';

import {
  useHistory,
} from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

export default function SignIn({ onSubmit }) {
  const classes = useStyles();
  const history = useHistory();
  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Typography component="h1" variant="h5">
          Login
        </Typography>
        <form
          className={classes.form}
          onSubmit={e => {
            e.preventDefault();
            onSubmit({
              userId: e.target.userId.value,
              nickname: e.target.nickname.value,
              theme: e.target.theme.checked ? 'dark' : 'light',
              useCustomQuery: e.target.useCustomQuery.checked,
            });
            history.push('/chat');
          }}
        >
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="userId"
            label="User Id"
            name="userId"
            autoFocus
          />
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="nickname"
            label="Nick Name"
            id="nickname"
          />
          <FormControlLabel
            control={<Checkbox value="dark" color="primary" name="theme" />}
            label="Apply dark theme"
          />
          <FormControlLabel
            control={<Checkbox value="customQuery" color="primary" name="useCustomQuery" />}
            label="Use custom user list"
          />
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            Start
          </Button>
        </form>
      </div>
    </Container>
  );
}
