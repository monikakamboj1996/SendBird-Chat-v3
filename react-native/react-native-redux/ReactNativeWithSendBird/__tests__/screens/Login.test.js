import 'react-native';
import React from 'react';
import Login from '../../src/screens/Login';
import store from '../../src/store';
import renderer from 'react-test-renderer';

describe('screen/Login', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Login store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
