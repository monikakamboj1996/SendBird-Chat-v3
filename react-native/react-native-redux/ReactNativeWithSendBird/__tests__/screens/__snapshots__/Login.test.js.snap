// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen/Login renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 5,
        "marginTop": 35,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/icon_sb_512.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#7d62d9",
          "fontSize": 30,
          "fontWeight": "600",
        }
      }
    >
      SendBird
    </Text>
    <Text
      style={
        Object {
          "color": "#7d62d9",
          "fontSize": 13,
          "fontWeight": "500",
        }
      }
    >
      React Native
    </Text>
  </View>
  <View
    style={
      Object {
        "borderColor": "#ddd",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginLeft": 28,
        "marginRight": 28,
        "marginTop": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      duration={100}
      label="User ID"
      maxLength={16}
      onChangeText={[Function]}
      placeholder="User ID"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "fontSize": 13,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "borderColor": "#ddd",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginLeft": 28,
        "marginRight": 28,
        "marginTop": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      duration={100}
      label="Nickname"
      maxLength={16}
      onChangeText={[Function]}
      placeholder="Nickname"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "fontSize": 13,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 50,
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#6e5baa",
              },
              false,
              false,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  false,
                ],
              ]
            }
          >
            CONNECT
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#e03131",
        "fontSize": 12,
      }
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "marginTop": 15,
          "paddingLeft": 28,
          "paddingRight": 28,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#8e8e8e",
          "fontSize": 12,
        }
      }
    >
      Sample UI v3.0.0 / SDK v.3.0.99
    </Text>
  </View>
</View>
`;
